ospf6d/ospf6_message.c

```diff
@@ -2037,7 +2037,7 @@ int ospf6_lsupdate_send_neighbor(struct thread *thread)
 
 	/* lsupdate_list lists those LSA which doesn't need to be
 	   retransmitted. remove those from the list */
-	for (ALL_LSDB(on->lsupdate_list, lsa)) {
+	for (ALL_LSDB(on->lsupdate_list, lsa, lsanext)) {
```

Comment:
It seems `ospf6d` is `assert`()ing in this part on the topology test `bfd-profiles-topo1`.

I just compiled and ran it and it is 100% reproducible:

```
ospf6d: ospf6d/ospf6_message.c:2070: ospf6_lsupdate_send_neighbor: Assertion `lsa->lock == 2' failed.

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50      ../sysdeps/unix/sysv/linux/raise.c: Arquivo ou diretório inexistente.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff6bd5535 in __GI_abort () at abort.c:79
#2  0x00007ffff6bd540f in __assert_fail_base (fmt=0x7ffff6d37ee0 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=0x55555564fda0 "lsa->lock == 2", file=0x55555564d860 "ospf6d/ospf6_message.c", line=2070, 
    function=<optimized out>) at assert.c:92
#3  0x00007ffff6be3102 in __GI___assert_fail (assertion=0x55555564fda0 "lsa->lock == 2", file=0x55555564d860 "ospf6d/ospf6_message.c", line=2070, 
    function=0x555555651060 <__PRETTY_FUNCTION__.18581> "ospf6_lsupdate_send_neighbor") at assert.c:101
#4  0x00005555555e36ee in ospf6_lsupdate_send_neighbor (thread=0x7fffffffe3d0) at ospf6d/ospf6_message.c:2070
#5  0x00007ffff70752aa in thread_call (thread=0x7fffffffe3d0) at lib/thread.c:1581
#6  0x00007ffff6fb3885 in frr_run (master=0x612000000040) at lib/libfrr.c:1096
#7  0x000055555558d560 in main (argc=5, argv=0x7fffffffe5f8, envp=0x7fffffffe628) at ospf6d/ospf6_main.c:230
(gdb) f 4
#4  0x00005555555e36ee in ospf6_lsupdate_send_neighbor (thread=0x7fffffffe3d0) at ospf6d/ospf6_message.c:2070
2070                    assert(lsa->lock == 2);
(gdb) l
2065                    ospf6_lsa_age_update_to_send(lsa, on->ospf6_if->transdelay);
2066                    memcpy(p, lsa->header, OSPF6_LSA_SIZE(lsa->header));
2067                    p += OSPF6_LSA_SIZE(lsa->header);
2068                    lsa_cnt++;
2069
2070                    assert(lsa->lock == 2);
2071                    ospf6_lsdb_remove(lsa, on->lsupdate_list);
2072            }
2073
2074            if (lsa_cnt) {
(gdb) p *lsa
$1 = {name = "[Link Id:0.0.0.3 Adv:10.254.254.4]", '\000' <repeats 29 times>, rn = 0x60b0000161a0, lock = 1 '\001', flag = 0 '\000', birth = {tv_sec = 47278, tv_usec = 561291}, originated = {tv_sec = 0, 
    tv_usec = 0}, received = {tv_sec = 0, tv_usec = 0}, installed = {tv_sec = 47278, tv_usec = 561303}, expire = 0x0, refresh = 0x0, retrans_count = 0, lsdb = 0x60400002d550, header = 0x60600001f7c0}
```

Reply:
fixed